StackExchange.Redis/StackExchange/Redis/RedisDatabase.cs

```diff
@@ -45,6 +46,37 @@ public RedisValue DebugObject(RedisKey key, CommandFlags flags = CommandFlags.No
             return ExecuteSync(msg, ResultProcessor.RedisValue);
         }
 
+        public bool GeoAdd(RedisKey key, RedisValue value0,RedisValue value1,RedisValue value2,CommandFlags flags = CommandFlags.None)
+        {
+            var msg = Message.Create(Database, flags, RedisCommand.GEOADD, key, value0,value1,value2);
+            return ExecuteSync(msg, ResultProcessor.Boolean);
+        }
+
+        public long GeoAdd(RedisKey key, RedisValue[] values, CommandFlags flags = CommandFlags.None)
+        {
+            if(values.Length%3!=0) throw new Exception("Each value must provide latitude, longitude and name");
+            var msg = Message.Create(Database,flags, RedisCommand.GEOADD, key, values);
+            return ExecuteSync(msg, ResultProcessor.Int64);
+        }
+
+        public RedisValue GeoDist(RedisKey key, RedisValue value0, RedisValue value1, GeoUnit geoUnit = GeoUnit.m,CommandFlags flags = CommandFlags.None)
+        {
+            var msg = Message.Create(Database, flags, RedisCommand.GEODIST, key,value0,value1,geoUnit.ToString());

```

Comment:
if the enum names change, this will need a map somewhere - probably just a `private static RedisValue[] geoUnitNames = {"m", ...};` - but in either case: what if it isn't one of the defined values? (enums are not checked)